packages/react-hot-loader/src/utils.dev.js

```diff
@@ -0,0 +1,76 @@
+import React, { Component } from 'react'
+import hoistNonReactStatic from 'hoist-non-react-statics'
+import AppContainer from './AppContainer.dev'
+
+const getDisplayName = WrappedComponent =>
+  WrappedComponent.displayName || WrappedComponent.name || 'Component'
+
+const copyReactProp = (source, target) => {
+  hoistNonReactStatic(target, source)
+  target.displayName = `HotExported${getDisplayName(source)}`
+  target.WrappedComponent = source
+
+  target.propTypes = source.contextTypes
+  target.childContextTypes = source.childContextTypes
+  target.contextTypes = source.contextTypes
+}
+
+const makeHotExport = (sourceModule, getInstances) => {
+  const updateInstances = () => {
+    getInstances().forEach(inst => inst.forceUpdate())
+  }
+
+  const thenUpdateInstances = () => Promise.resolve(true).then(updateInstances)
+
+  if (sourceModule.hot) {
+    sourceModule.hot.accept(() => {
+      // Mark as self-accepted for Webpack
+      // Update instances for parsel
+      thenUpdateInstances()
+    })
+
+    // webpack way
+    if (sourceModule.hot.addStatusHandler) {
+      if (sourceModule.hot.status() === 'idle') {
+        sourceModule.hot.addStatusHandler(status => {
+          if (status === 'apply') {
+            thenUpdateInstances()
+          }
+        })
+      }
+    }
+  }
+}
+
+export const hotExported = (sourceModule, WrappedComponent, props = {}) => {
+  let instances = []
+
+  class ExportedComponent extends Component {
```

Comment:
We have to [wrap displayName](https://github.com/neoziro/recompact/blob/master/src/wrapDisplayName.js). It is a well known pattern in hoc and it is better for debugging.

Reply:
It is performed on line 10, but I should use some helper to do it.
PS: And I should not transfer contextTypes. Have no idea why I did it.